composer/datasets/evaluator.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import List
+
+import yahp as hp
+from torchmetrics.classification.accuracy import Accuracy
+from torchmetrics.classification.f_beta import F1
+from torchmetrics.classification.matthews_corrcoef import MatthewsCorrcoef
+from torchmetrics.collections import MetricCollection
+from torchmetrics.metric import Metric
+from torchmetrics.regression.pearson import PearsonCorrcoef
+from torchmetrics.regression.spearman import SpearmanCorrcoef
+
+from composer.core.types import Metrics
+from composer.datasets.dataset_registry import get_dataset_registry
+from composer.datasets.hparams import DataloaderSpec, DatasetHparams
+from composer.models.loss import CrossEntropyLoss
+
+log = logging.getLogger(__name__)
+
+
+@dataclass
+class EvaluatorHparams(hp.Hparams):
+    hparams_registry = {  # type: ignore
+        "eval_dataset": get_dataset_registry(),
+    }
+
+    metric_registry = {
+        "Accuracy": Accuracy,
+        "MatthewsCorrcoef": MatthewsCorrcoef,
+        "F1": F1,
+        "PearsonCorrcoef": PearsonCorrcoef,
+        "SpearmanCorrcoef": SpearmanCorrcoef,
+        "CrossEntropyLoss": CrossEntropyLoss,
+    }
+
+    label: str = hp.required(doc="label")
+    eval_dataset: DatasetHparams = hp.required(doc="Training dataset hparams")
+    metrics: List[str] = hp.required(doc="metrics", template_default=list)
+
+    def initialize_object(self):
+        dataset = self.eval_dataset.initialize_object()
+
+        def resolve_metric_list(metric_names: List[str]) -> List[Metric]:
+            resolved_metrics = []
+            for metric_name in metric_names:
+                if metric_name in self.metric_registry:
+                    metric_instance = self.metric_registry[metric_name]()
+                    resolved_metrics.append(metric_instance)
+                else:
+                    log.warning(f"The metric {metric_name} is not a registered metric. Add it to the metric registry.")
```

Comment:
Should we raise an exception instead of a warning if a metric is not valid? I think I would want my training to crash before I check it the next day, but curious for other people's thoughts as well.